sui/its.js

```diff
@@ -501,27 +505,41 @@ async function linkCoin(keypair, client, config, contracts, args, options) {
         symbol,
         metadata,
         tokenType,
+        tokenManager === tokenManagerTypes[1] ? treasuryCap : null,
     );
 
-    if (!tokenId) throw new Error(`error resolving token id from registration tx, got ${tokenId}`);
-    if (!options.channel && !channelId) throw new Error(`error resolving channel id from registration tx, got ${channelId}`);
+    if (!tokenId) {
+        throw new Error(`error resolving token id from registration tx, got ${tokenId}`);
+    }
+    if (!options.channel && !channelId) {
+        throw new Error(`error resolving channel id from registration tx, got ${channelId}`);
+    }
 
     const channel = options.channel ? options.channel : channelId;
 
     // User then calls linkToken on ITS Chain A with the destination token address for Chain B.
     // This submits a LinkToken msg type to ITS Hub.
     txBuilder = new TxBuilder(client);
 
-    const tokenManagerType = await txBuilder.moveCall({
-        target: `${itsConfig.address}::token_manager_type::lock_unlock`,
-    });
+    // Token manager type
+    const tokenManagerType =
+        tokenManager == tokenManagerTypes[1]
+            ? await txBuilder.moveCall({
+                  target: `${itsConfig.address}::token_manager_type::mint_burn`,
+              })
+            : await txBuilder.moveCall({
+                  target: `${itsConfig.address}::token_manager_type::lock_unlock`,
+              });
```

Comment:
```suggestion
    const tokenManagerType = await txBuilder.moveCall({
        target: `${itsConfig.address}::token_manager_type::${tokenManager}`,
    })
```
confusing, simplify?